### DISCUSSION TITLE
Using Lazyterm as a ToggleTerm Replacement

### DISCUSSION THREAD

**Discussion by karimlevallois** [LazyVim/LazyVim#2602] [Q&A]
Hi everyone

I used to have a keycap in LunarVim that opened a floating terminal window and ran the Python script of the current buffer...

`map("n", "<leader>pt", ":cd %:p:h |:TermExec cmd='python3 %<cr>", { desc = "Run Python Script In Floating Terminal" })`

Is there a way to do similar in LazyTerm?

**Comment by mikavilpas** [comment]
You can run a command e.g. like this

```lua
require("lazyvim.util.terminal").open({"python3", "-v"})
```

**Reply by karimlevallois** [reply]
Thanks for the reply. I can get python to open in a floating window with...

`map("n", "<leader>pt", function() require("lazyvim.util.terminal").open({"python3"}) end, { desc = "Run Python Script In Floating Terminal" })`

but cannot get it to run the current buffer with...

`map("n", "<leader>pt", function() require("lazyvim.util.terminal").open({"python3 %"}) end, { desc = "Run Python Script In Floating Terminal" })`

**Reply by mikavilpas** [reply]
I think you need to get the full path of the current buffer and give it to python:

```lua
vim.keymap.set("n", "<leader>TT", function()
  local absolutePath = vim.api.nvim_buf_get_name(0)
  require("lazyvim.util.terminal").open({ "python3", absolutePath })
end, { desc = "Run Python Script In Floating Terminal" })
```

I don't know what would be a good way to handle error cases, you may run into some issues with this.

You could also look into https://github.com/akinsho/toggleterm.nvim which is a more full featured terminal experience - maybe it will work better for you.

**Reply by karimlevallois** [reply]
Yeah, it opens the file itself, but it seemed to crash the terminal when I tried to reference another file in the same root (script asks for a file and if nothing is entered it defaults to a certain file in the same folder as the script). Perhaps the terminal doesn’t stay in the file directory.

**Reply by mikavilpas** [reply]
Hmm it might be caused by many different issues. I have a setup that is working well, although I'm not sure I can explain everything here.

If you'd like to take a look, https://github.com/sp3ctum/dotfiles
